### ISSUE TITLE
Run cmd script from the editor

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#945]
Sometimes I run an external script after I build a game with the editor, usually for packaging purposes. It would be cool if the editor could call an external script. In my head I imagine like using cmd.exe with the working directory being the game project folder, so I could do like `pack.bat` to generate my game packages for Linux and Windows. I don't think I would actually want to run the script all the time so maybe being able to filter by target, like, only run when building for release.

**Comment by ivan-mogilko** [comment]
Perhaps AGS Editor needs a mechanism to setup compilation sequence, similar to other IDEs, where you select game compilation, and run external tools.

**Comment by ericoporto** [comment]
In keil uVision, it had a hook for Before Compile (e.g.for script that generated code, like a custom dialog to AGS Script parser) and After Build (used for deploying the build somewhere). It also created some environment variables when calling the script.

Other than this the release and debug switch would then be moved to somewhere easier to change, like the toolbar itself.

**Comment by morganwillcock** [comment]
If the script compiler and game packing is done by separate component(s) and not by the Editor, then all of this can probably run externally and in parallel. I would like to think that the media pipeline is incorporated here to, so if a music/video source is lossless and you want compressed versions this is handled and cached through the same build process (and gives you the chance to pack content differently depending on target platform as well as any flags that you specify).

**Comment by rofl0r** [comment]
> all of this can probably run externally and in parallel

indeed, if the AGSC compiler was an external command and not part of the editor, the whole build process could easily be packed into a makefile and parallelized.

**Comment by ivan-mogilko** [comment]
This won't be pretty if such ability is hacked into the editor as it is now, because there's no proper separate build settings, and no building control system with distinguishable steps to allow you insert custom commands anywhere, so this will end up as an another awkward option in project data again (like Build Target). (And what if someone finds they need to pass arguments into the script?)

At the same time it's known that Editor can run compilation from command line. There's a related ticket on fixing problems with that: #666

Question: is not solving #666 actually enough to solve running any kinds of scripts from command line, and thus render this ticket unnecessary?

**Comment by ivan-mogilko** [comment]
Of course, in the end all this boils down to standalone compilation tools.

Would it be better to cease adding build-related changes to the editor (unless they provide critical function which is too difficult to perform by other means), and instead focus on tool decoupling?
Then we could close this ticket, and maybe even #666 if we find it's too complicated to fix.

**Comment by ivan-mogilko** [comment]
I think I will close this, because we are definitely going towards separating build tools from the editor, which would be much more flexible approach. And this proposed feature is not a critical one at the moment.
